### ISSUE TITLE
[Cahill-Keyes] Conformal version

### ISSUE THREAD

**Issue by Fil** [d3/d3-geo-polygon#16]
We could use the Schwarz-Christoffel transformation. http://www.genekeyes.com/CAHILL-VARIANTS/Cahill-Conformal.html shows one made by @jrus

**Comment by jrus** [comment]
Yeah, I have been intending for a long time to properly implement this in D3. The original [SCToolbox](http://www.math.udel.edu/~driscoll/SC/) version I made was outrageously slow to compute, so I let my computer run for a couple days finding the positions of a pretty dense grid of Chebyshev nodes in latitude/longitude for half an octant, and used [Chebfun](http://www.chebfun.org) to efficiently compress and evaluate that.

It would in general be a good idea to implement a few parts from the guts of Chebfun in JavaScript, and use those for many map projections. It should in many cases dramatically speed up computations, especially if the inner loops can be implemented as wasm.

I've had a lot of trouble getting time to work on this kind of project in the past year, as I've been taking care of a (now 20 month old) baby full time. But I still do intend to get something up sometime. I intend to properly write a paper about this projection at some point, to submit to some kind of cartography conference/journal. And there are a couple related map projections which I would like to produce (both an equal-area version, and one optimizing for distance errors), but which will take some significant research effort to figure out.

Not sure it would be reasonable to call this particular projection "Cahill–Keyes" though. I would call it something like the "Conformal Octahedral Butterfly" projection. (I don't personally like sticking my own name on things, but also wouldn’t really be comfortable sticking someone else’s name on a project I spent a few months figuring out the details for.)

**Comment by Fil** [comment]
@jrus has implemented a few possibilities at https://observablehq.com/@jrus/conformal-octahedron

(but I can't find the link to the version with corners, à la Cahill.)

**Comment by jrus** [comment]
There was never any such version until I spent a month of effort making one up in 2015. I still haven’t ported it to Javascript though.
